phpdotnet/phd/Package/PHP/Web.php

```diff
@@ -234,35 +234,75 @@ public function footer($id) {
 
     protected function writeJsonIndex() {
         v("Writing search indexes..", VERBOSE_FORMAT_RENDERING);
-        $ids = array();
-        $desc = array();
+        [$entries, $descriptions] = $this->processIndexForSearch();
+        file_put_contents(
+            $this->getOutputDir() . "search-index.json",
+            json_encode($entries)
+        );
+        file_put_contents(
+            $this->getOutputDir() . "search-description.json",
+            json_encode($descriptions)
+        );
+        v("Index written", VERBOSE_FORMAT_RENDERING);
+    }
+
+    /**
+     * Processes the search index to extract entries and descriptions.
+     */
+    private function processIndexForSearch(): array {
```

Comment:
Nit: this is only for generating entries for the JSON file used in the JS-based search function on `php.net` and the method name should probably reflect that (eg. `processIndexForJsonIndex()` / `processIndexForJsonSearch()` / something else (I'm not good at naming things)).

Reply:
Yup, makes sense. I like the suggested names, but went with `processJsonIndex` that is also good and stays in line with the existing `writeJsonIndex` method. Also added this context to the docblock. What do you think?

Reply:
Looks good to me. Thanks.